client-proxy/main.go

```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+	"github.com/OpenLineage/OpenLineage/client-proxy/config"
+	"github.com/OpenLineage/OpenLineage/client-proxy/consumer"
+	LineageController "github.com/OpenLineage/OpenLineage/client-proxy/controllers/lineage"
+	"github.com/OpenLineage/OpenLineage/client-proxy/database"
+	LineageService "github.com/OpenLineage/OpenLineage/client-proxy/services/lineage"
+	"github.com/OpenLineage/OpenLineage/client-proxy/transports"
+	"github.com/OpenLineage/OpenLineage/client-proxy/validator"
+	"github.com/gin-gonic/gin"
+	"log"
+)
+
+func main() {
+	conf, err := config.Load(".")
+	if err != nil {
+		log.Fatalln("Cannot load config:", err)
+	}
+
+	db := database.New(conf)
+
+	failedEventHandler, err := validator.NewFailedEventHandler(conf)
+	if err != nil {
+		log.Fatalln(err)
+	}
+	lineageService := LineageService.New(db, failedEventHandler)
+	lineageController := LineageController.New(lineageService)
+
+	transport, err := transports.Create(conf)
+	if err != nil {
+		log.Fatalln(err)
+	}
+	lineageEventConsumer := consumer.New(db, transport)
+	go lineageEventConsumer.Run()
+
+	router := gin.Default()
+	router.Use(gin.Logger())
+	router.Use(gin.Recovery())
+	router.POST("/api/v1/lineage", lineageController.PostLineage)
+
+	log.Fatalln(router.Run(":8080"))
```

Comment:
Do we need to handle the shutdown case? What if we have OL events queued?

Reply:
The queued events are in the sqlite3 database as well as the checkpoint. This is designed to be able to restart the app in case of crash.

Reply:
I see, that's part of the design. Would be helpful to document that and have metrics around the `#` of queued events, etc.

Reply:
Yeah, it's not in this skeleton yet, but the design #1412 covers the metrics publication.